packages/now-cli/src/util/dev/server.ts

```diff
@@ -1104,23 +1104,24 @@ export default class DevServer {
    */
   getNowProxyHeaders(
     req: http.IncomingMessage,
-    nowRequestId: string
+    nowRequestId: string,
+    xfwd: boolean
   ): http.IncomingHttpHeaders {
     const ip = this.getRequestIp(req);
     const { host } = req.headers;
-    return {
-      ...req.headers,
-      Connection: 'close',
-      'x-forwarded-host': host,
-      'x-forwarded-proto': 'http',
-      'x-forwarded-for': ip,
+    const headers: http.IncomingHttpHeaders = {
+      connection: 'close',
       'x-real-ip': ip,
-      'x-now-trace': 'dev1',
-      'x-now-deployment-url': host,
-      'x-now-id': nowRequestId,
-      'x-now-log-id': nowRequestId.split('-')[2],
-      'x-zeit-co-forwarded-for': ip,
+      'x-vercel-deployment-url': host,
+      'x-vercel-forwarded-for': ip,
+      'x-vercel-id': nowRequestId,
     };
+    if (xfwd) {
```

Comment:
Why do you need to conditionally add these 3 headers?

Reply:
Cause of `node-http-proxy` setting these headers as well (in the case of `startDevServer()`). Some of the `x-forwarded-*` headers are a comma separated list, so setting them here first causing the values to be reported twice (i.e. `x-forwarded-for: 127.0.0.1,127.0.0.1`).

Reply:
Can we confirm this works correctly with non-node runtimes? For example, add a check to the existing python dev test.

Reply:
Is the idea to test a runtime that _doesn't_ implement `startDevServer()`?

Reply:
Indeed, to make sure both return the correct headers 👍

Reply:
Added in 4c672594783268477e270b69444da7cf0c2b857b, though `startDevServer()` may end up being implemented in `@vercel/python` at some point.

Reply:
You could also add a test to the `validateResponseHeaders()` function which is used in multiple places

Reply:
That's different though since it's testing for the final _response_ headers, rather than the request header that the API endpoint receives.

Reply:
I'm not so sure? See [line 153](https://github.com/vercel/vercel/pull/4850/files#diff-948deac20e5be5b5293a06cead4b7c55R153)